### PR TITLE
[TS] LPS-158932 Allow for userID to be passed correctly when using Experiences with Workflow

### PR DESCRIPTION
**Pull request by ericyanLr**
Hi Echo team,

Can you help review this PR?

Please let us know if you have any questions.
Thanks!

**Notes from @jakesliwka:**
> https://issues.liferay.com/browse/LPS-158932
> 
> While having workflow enabled and creating a new page with multiple experiences, once approved, the pages would appear as blank.
> 
> This was because only when workflow was enabled, and updateLayoutPageTemplateStructureData was called, a user ID of 0 was passed passed in to the _layoutPageTemplateStructureRelLocalService.addLayoutPageTemplateStructureRel method which threw an error saying that there cannot be a userId of 0. This caused the experiences to not display their changes made, despite being 'approved'. When tested with workflow disabled, the correct userID gets passed each time.
> 
> The userID was being retrieved with PrincipalThreadLocal.getUserId() . The solution was to go back to when we had a valid userID and pass it along to make sure that a correct one is used and set it back afterwards, as well as ensure that it is not used anywhere else during the process.
> 
> We decided to pass in a valid userId to _layoutCopyHelper.copyLayout(draftLayout, layout) located in LayoutWorkflowHandler.java, which ensures that these changes are only called when workflow is enabled.
> 
> Testing resulted in confirming that with the applied changes, the behavior is very similar to publishing a page without workflow, passing the correct userId's when needed.
> 
> The worry was that if workflow and publications were to be eventually integrated, this issue could potentially pass the wrong userId.
> This [PTR](https://issues.liferay.com/browse/PTR-3253) confirmed that since workflow does not work with publications, this fix can proceed safely.

/cc @noornajj

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by ericyanLr** [comment]
ci:test:sf

**Comment by ericyanLr** [comment]
ci:test:relevant

**Comment by ericyanLr** [comment]
ci:reevaluate:1317401_4564

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by michaelhashimoto** [comment]
ci:reevaluate:1277401_4469

**Comment by lfbesada** [comment]
ci:test:relevant

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/workflow/LayoutWorkflowHandler.java [hunk 1]
```diff
@@ -106,12 +107,19 @@ public Layout updateStatus(
 
 		Layout draftLayout = layout.fetchDraftLayout();
 
+		long originalUserId = PrincipalThreadLocal.getUserId();
+
+		PrincipalThreadLocal.setName(userId);
+
```
Comment (lfbesada) [inline modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/workflow/LayoutWorkflowHandler.java:113]: This should be inside the try

**Comment by lfbesada** [comment]
Hi @ericyanLr
Thanks for the pull request!
I have left a review comment in case that helps but we cannot accept this PR as the reported bug is not reproducible in the current master.
Regards

cc:  @jakesliwka @noornajj @michaelhashimoto

**Comment by lfbesada** [comment]
see: https://github.com/liferay-echo/liferay-portal/pull/9337

**Pull request closed without merging**